### PR TITLE
Allow path variables in modules.d files

### LINKED ISSUE #13184: [Filebeat] path variables cannot be used in files from modules.d
Using `${path.config}` in files from `modules.d/` causes a somewhat ambiguous error.

> Exiting: 1 error: Error unpacking module config: error unpacking module googlecloud in a dict: missing field accessing ‘0.vpcflow’ 

The error doesn't mention that it failed to expand `${path.config}` in

```
- module: googlecloud
  vpcflow:
    enabled: true
    var.project_id: my-gcp-project-id
    var.topic: googlecloud-vpc-flowlogs
    var.subscription_name: filebeat-googlecloud-vpc-flowlogs-su
    var.credentials_file: ${path.config}/gcp-service-account-xyz.json
```

### PR DESCRIPTION
**Pull request by andrewkroh**
This allows path variables (e.g. `${path.config}`) to be used in files loaded from modules.d.

Fixes #13184

### CODE REVIEW

File: filebeat/fileset/modules.go [hunk 1]
```diff
@@ -168,7 +173,7 @@ func mcfgFromConfig(cfg *common.Config) (*ModuleConfig, error) {
 
 	mcfg.Filesets = map[string]*FilesetConfig{}
 	for name, filesetConfig := range dict {
-		if name == "module" || name == "enabled" {
+		if name == "module" || name == "enabled" || name == "path" {
```
Comment (jsoriano) [inline filebeat/fileset/modules.go]: Why is this change needed?

Comment (andrewkroh) [reply filebeat/fileset/modules.go]: I order to allow ufcg to resolve the `$path.config` variables they must be part of the configuration. So after merging them in you end up with something like:

```
  module: xyz
  enabled: true
  dataset1:
    var.something: ${path.config}/some/resource.csv
  dataset2:
    enabled: true
    var.foo: bar
  path:
    home: /usr/share/filebeat
    config: /etc/filebeat
    data: y
    logs: z
```

Comment (jsoriano) [reply filebeat/fileset/modules.go]: Oh I see :+1:
